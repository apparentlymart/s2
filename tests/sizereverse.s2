#-*-perl-*-

layerinfo type = core;
layerinfo "majorversion" = 0;

function builtin string(int i) : string;
function builtin int(string s) : int;

function main ()
{
    var int[] iarray;
    $iarray[0] = 0;
    $iarray[1] = 1;
    $iarray[2] = 2;
    $iarray[3] = 3;
    $iarray[4] = 4;
    $iarray[5] = 5;

    $iarray = reverse $iarray;
    $iarray[0] = 0;
    $iarray[5] = 5;
    $iarray = reverse $iarray;
    
    println "Size of array = " + string(size $iarray);
    println "Iterating over array:";
    foreach var int i ($iarray)
    {
	println $i;
    }

    var string mystr = "hello";
    var string mystr_backwards = reverse $mystr;
    println $mystr_backwards;

    println "Done.";
}
