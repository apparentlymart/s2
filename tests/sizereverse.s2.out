Size of array = 6
Iterating over array:
5
1
2
3
4
0
Done.
